statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,538 @@
+        of ones is constructed
+    cluster : array-like or None
+        Cluster for each observation. If none, an array
+        of ones is constructed
+    weights : array-like or None
+        The weight for each observation. If none, an array
+        of ones is constructed
+    nest : boolean
+        allows user to specify if PSU's with the same
+        PSU number in different strata are treated as distinct PSUs.
+
+    Attributes
+    ----------
+    weights : (n, ) array
+        The weight for each observation
+    nstrat : integer
+        The number of district strata
+    sclust : (n, ) array
+        The relabeled cluster array from 0, 1, ..
+    strat : (n, ) array
+        The related strata array from 0, 1, ...
+    ncs : (self.nstrat, ) array
+        Holds the number of clusters in each stratum
+    sfclust : ndarray
+        The stratum for each cluster
+    nclust : integer
+        The total number of clusters across strata
+    """
+
+    def __init__(self, strata=None, cluster=None, weights=None, rep_weights=None, nest=True):
+        strata, cluster, self.weights = self._check_args(strata, cluster, weights)
+        self.rep_weights = rep_weights
+        # Recode strata and clusters as integer values 0, 1, ...
+        _, self.strat = np.unique(strata, return_inverse=True)
+        _, clust = np.unique(cluster, return_inverse=True)
+
+        # the number of distinct strata
+        self.nstrat = max(self.strat) + 1
+
+        # If requested, recode the PSUs to be sure that the same PSU number in
+        # different strata are treated as distinct PSUs.  This is the same as
+        # the nest option in R.
+        if nest:
+            m = max(clust) + 1
+            sclust = clust + m*self.strat
+            _, self.sclust = np.unique(sclust, return_inverse=True)
+        else:
+            self.sclust = clust.copy()
+
+        # The number of clusters per stratum
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.ncs = np.bincount(self.strat[ii])
+
+        # The stratum for each cluster
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.sfclust = self.strat[ii]
+
+        # The total number of clusters over all stratum
+        self.nclust = np.sum(self.ncs)
+
+    def __str__(self):
+        """
+        The __str__ method for our data
+        """
+        summary_list = ["Number of observations: ", str(len(self.strat)),
+                        "Sum of weights: ", str(self.weights.sum()),
+                        "Number of strata: ", str(self.nstrat),
+                        "Number of clusters per stratum:", str(self.ncs)]
+
+        return "\n".join(summary_list)
+        # print("Number of observations: ", len(self.strat))
+        # print("Sum of weight: ", self.weights.sum())
+        # print("Number of strata: ", self.nstrat)
+        # print("The number of clusters per stratum: ", self.ncs)
+
+    def _check_args(self, strata, cluster, weights):
```

Comment:
`_check_args` should also take `rep_weights` as an argument.  It would be OK to have `rep_weights` and everything else None.  We need to think through all the combinations of input arguments and determine which combinations are valid.